docs/signals/attributes/index.md

```diff
@@ -0,0 +1,110 @@
+---
+title: "Defining Attributes"
+sidebar_position: 20
+description: "Attributes."
+sidebar_label: "Attributes"
+---
+
+Attributes are the building blocks of Snowplow Signals. They represent specific facts about user behavior and are calculated based on events in your Snowplow pipeline. This guide explains how to define an `Attribute`, and use `Criteria` to filter events for precise data aggregation.
+
+- **Number of page views in the last 7 Days:** counts how many pages a user has viewed within the past week.
+- **Last product viewed:** identifies the most recent product a user interacted with.
+- **Previous purchases:** provides a record of the user's past transactions. 
+
+
+### Basic Usage
+
+An `Attribute` can be defined to count the number of pageviews in a session through the Python SDK as follows:
+
+```python
+from snowplow_signals import Attribute, Event
+
+page_views_attribute = Attribute(
+    name='page_views_count',
+    type='int32',
+    events=[
+        Event(
+            vendor="com.snowplowanalytics.snowplow",
+            name="page_view",
+            version="1-0-0",
+        )
+    ],
+    aggregation='counter'
+)
+
+```
+
+### Advanced Usage
+
+Add a `Criteria` to refine the events used to calculate the `Attribute`. For example, if you want to see the number of pageviews on a particular web page on your site.
+
+
+```python
+from snowplow_signals import Attribute, Event, Criteria, Criterion
+
+page_views_attribute = Attribute(
+    name='page_views_count',
+    type='int32',
+    events=[
+        Event(
+            vendor="com.snowplowanalytics.snowplow",
+            name="page_view",
+            version="1-0-0",
+        )
+    ],
+    aggregation='counter',
+    filter=Criteria(
+        all=[
+            Criterion(
+                property="page_title",
+                operator="=",
+                value="home_page",
+            ),
+        ],
+    ),
+)
+
+```
+
+The table below lists all arguments for an `Attribute`:
+
+| **Argument Name** | **Description** | **Type** |
+| --- | --- | --- | 
+| `name` | The name of the Attribute | `string` |
+| `description` | The description of the Attribute | `string` |
+| `type` | The type of the aggregation | One of: `bytes`, `string`, `int32`, `int64`, `double`, `float`, `bool`, `unix_timestamp`, `bytes_list`, `string_list`, `int32_list`, `int64_list`, `double_list`, `float_list`, `bool_list`, `unix_timestamp_list`,  |
+| `tags` | Metadata for the Attribute | |
+| `events` | List of Snowplow Events that the Attribute is calculated on | List of `Event` type |
+| `aggregation` | The aggregation type of the Attribute  | One of:  `counter`, `sum`, `min`, `max`, `mean`, `first`, `last`, `unique_list` |
+| `property_syntax` | The syntax used to reference the property. | One of: `snowflake`, `blobl` |
```

Comment:
We can remove this from the docs (also in the criteria) as we don't want users to set this.